### PR TITLE
refactor assignment 2

### PR DESCRIPTION
**Pull request by samueleceroni**

### CODE REVIEW

**Comment by samueleceroni** [comment]
I tested it manually and it seems to work for me

File: Assignment2/superserver.c [hunk 1]
```diff
@@ -280,72 +290,90 @@ void handle_service(int inputSocketFD, ServiceData *config, char * const envp[])
...
 	
-	FD_ZERO(&socketsSet);
-	for (size_t i = 0; i < config.size; i++) {
-		printf("socket set %d\n", i);
-		FD_SET(config.services[i].socketFD, &socketsSet);
-	}
+	initialize_socket_set();
 
-	// Handle signals sent by son processes
+	// HandleEXIT_SUPERSERVER_CONFIG_FILE_ERROR signals sent by son processes
```
Comment (Carlovan) [inline Assignment2/superserver.c]: I think this got copied accidentally ;-)

File: Assignment2/superserver.c [hunk 2]
```diff
@@ -280,72 +290,90 @@ void handle_service(int inputSocketFD, ServiceData *config, char * const envp[])
...
 
+void initialize_socket_set(){
+	FD_ZERO(&socketsSet);
+	for (size_t i = 0; i < config.size; i++) {
+		printf("socket set %d\n", i);
+		FD_SET(config.services[i].socketFD, &socketsSet);
+	}
+}
+
+void main_loop_one_cycle(int highestFd, char **env){
```
Comment (Carlovan) [inline Assignment2/superserver.c]: I don't like this name... maybe we could just call this `main_loop` and move the `while` inside

File: Assignment2/superserver.c [hunk 3]
```diff
@@ -280,72 +290,90 @@ void handle_service(int inputSocketFD, ServiceData *config, char * const envp[])
...
+ServiceDataVector read_server_configuration(){
+	FILE *fp = fopen(SUPERSERVER_CONF_FILE_NAME, "r");
+	if(fp == NULL)
+		die(EXIT_SUPERSERVER_CONFIG_FILE_ERROR);
+	ServiceDataVector config = read_configuration(fp);
+	fclose(fp);
+	return config;
+}
+
+void handle_positive_bit_select(ServiceData* config, char **env, fd_set* socketsSet){
```
Comment (Carlovan) [inline Assignment2/superserver.c]: I don't like this name... We could rename it to `handle_selected_service` or maybe to `handle_service` and then change the current `handle_service` function to `spawn_service`

File: Assignment2/superserver.c [hunk 4]
```diff
@@ -280,72 +290,90 @@ void handle_service(int inputSocketFD, ServiceData *config, char * const envp[])
...
+}
+
+
 //Function prototype devoted to handle the death of the son process
 void handle_signal (int sig);
 
 ServiceDataVector config;
 fd_set socketsSet;
 
+void initialize_socket_set(){
```
Comment (Carlovan) [inline Assignment2/superserver.c]: I would pass the set pointer as a parameter to be consistent with other functions (like `handle_positive_bit_select`)

**Review by Carlovan — CHANGES_REQUESTED** [review]

**Pull request merged**